uitest/Extensions/WindowsDriverExtensions.cs

```diff
@@ -0,0 +1,139 @@
+    /// <param name="pollingIntervalInMS">Action interval in milliseconds</param>
+    /// <returns>Instance of <see cref="DefaultWait{T}"/></returns>
+    public static DefaultWait<WindowsDriver<WindowsElement>> Wait(
+        this WindowsDriver<WindowsElement> driver,
+        int timeoutInMS = DefaultTimeoutInMS,
+        int pollingIntervalInMS = DefaultPollingIntervalInMS)
+    {
+        return driver.Wait(TimeSpan.FromMilliseconds(timeoutInMS), TimeSpan.FromMilliseconds(pollingIntervalInMS));
+    }
+
+    /// <summary>
+    /// Create an instance of <see cref="DefaultWait{T}"/> which ignores all <see cref="WebDriverException"/> exceptions
+    /// </summary>
+    /// <param name="driver">Driver instance</param>
+    /// <param name="timeoutInMS">Wait timeout in milliseconds</param>
+    /// <param name="pollingIntervalInMS">Action interval in milliseconds</param>
+    /// <returns>Instance of <see cref="DefaultWait{T}"/></returns>
+    public static DefaultWait<WindowsDriver<WindowsElement>> WaitAndIgnoreExceptions(
+        this WindowsDriver<WindowsElement> driver,
+        int timeoutInMS = DefaultTimeoutInMS,
+        int pollingIntervalInMS = DefaultPollingIntervalInMS)
+    {
+        var wait = driver.Wait(timeoutInMS, pollingIntervalInMS);
+        wait.IgnoreExceptionTypes(typeof(WebDriverException));
+        return wait;
+    }
+
+    /// <summary>
+    /// Wait until an element is invisible or timeout
+    /// </summary>
+    /// <param name="driver">Driver instance</param>
+    /// <param name="by">Element locator</param>
+    /// <param name="timeoutInMS">Wait timeout in milliseconds</param>
+    /// <param name="pollingIntervalInMS">Action interval in milliseconds</param>
+    public static void WaitUntilInvisible(
+        this WindowsDriver<WindowsElement> driver,
+        By by,
+        int timeoutInMS = DefaultTimeoutInMS,
+        int pollingIntervalInMS = DefaultPollingIntervalInMS)
+    {
+        driver
+            .Wait(timeoutInMS, pollingIntervalInMS)
+            .Until(_ =>
+            {
+                try
+                {
+                    return !driver.FindElement(by).Displayed;
+                }
+                catch (WebDriverException)
+                {
+                    return true;
+                }
+            });
+    }
+
+    /// <summary>
+    /// Wait until an element is visible or timeout
+    /// </summary>
+    /// <param name="driver">Driver instance</param>
+    /// <param name="by">Element locator</param>
+    /// <param name="timeoutInMS">Wait timeout in milliseconds</param>
+    /// <param name="pollingIntervalInMS">Action interval in milliseconds</param>
+    public static WindowsElement WaitUntilVisible(
+        this WindowsDriver<WindowsElement> driver,
+        By by,
+        int timeoutInMS = DefaultTimeoutInMS,
+        int pollingIntervalInMS = DefaultPollingIntervalInMS)
+    {
+        return driver
+            .Wait(timeoutInMS, pollingIntervalInMS)
```

Comment:
Could it help to add logging to these actions? Like "waiting for X to be invisible"